app/views/components/_banner.html.erb

```diff
@@ -0,0 +1,28 @@
+<%
+  title ||= false
+  aside ||= false
+%>
+
+<% if aside %>
+ <div class="app-c-banner app-c-banner--grid">
+  <div class="grid-row app-c-banner__aside">
+    <div class="column-third">
+      <p><%= aside %></p>
+    </div>
+
+    <div class="column-two-thirds app-c-banner__main">
+      <h2 class="app-c-banner__title"><%= title %></h2>
+      <p class="app-c-banner__desc"><%= text  %></p>
+    </div>
+  </div>
+ </div>
+<% else %>
+  <div class="app-c-banner">
+    <% if title %>
+      <h2 class="app-c-banner__title"><%= title %></h2>
```

Comment:
You could probably have a block at the top for title and description to avoid repetition:

```erb
<% title_block = capture do %>
  <h2 class="app-c-banner__title"><%= title %></h2>
<% end %>
```

Then
```erb
<div class="column-two-thirds app-c-banner__main">
  <%= title_block %>
  <p class="app-c-banner__desc"><%= text  %></p>
</div>
```